navigation/app/src/main/java/com/mapbox/services/android/navigation/testapp/activity/navigationui/NavigationViewActivity.java

```diff
@@ -0,0 +1,44 @@
+package com.mapbox.services.android.navigation.testapp.activity.navigationui;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.widget.Button;
+
+import com.mapbox.services.android.navigation.testapp.R;
+import com.mapbox.services.android.navigation.ui.v5.NavigationView;
+import com.mapbox.services.android.navigation.v5.navigation.NavigationConstants;
+
+public class NavigationViewActivity extends AppCompatActivity {
+
+  @Override
+  protected void onCreate(@Nullable Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_navigation_view);
+
+    Button launchNavigationBtn = (Button) findViewById(R.id.launchBtn);
+    launchNavigationBtn.setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View v) {
+        launchNavigation();
+      }
+    });
+  }
+
+  private void launchNavigation() {
+    Intent navigationView = new Intent(this, NavigationView.class);
+    Bundle coordinates = new Bundle();
+    // Washington, D.C.
+    coordinates.putDouble(NavigationConstants.NAVIGATION_VIEW_ORIGIN_LNG_KEY, -77.009003);
```

Comment:
I assume the users required to use these constant keys for the bundled value? It might be better creating a `IntentBuilder` which takes in a `latlng` for the origin and destination and you breakdown that item in the background. See how the [PlacePicker](https://developers.google.com/places/android-api/placepicker) does this.

Reply:
`NavigationLauncher` addresses this confusion for now in https://github.com/mapbox/mapbox-navigation-android/pull/182/commits/0bd9490d8f4b106883029e554fa08a22432c3c34 - but ticketed building out a proper IntentBuilder here as well https://github.com/mapbox/mapbox-navigation-android/issues/202